glyphs2fontir/src/erase_open_corners.rs

```diff
@@ -363,14 +372,48 @@ fn seg_seg_intersection(seg1: PathSeg, seg2: PathSeg) -> Option<Intersection> {
                 t0: hit.segment_t,
                 t1: hit.line_t,
             }),
-        (bez0, bez1) => curve_curve_intersection_py(bez0, bez1),
+        (bez0, bez1) => return curve_curve_intersection_py(bez0, bez1),
+    }?;
+    if let (PathSeg::Line(l1), PathSeg::Line(l2)) = (seg1, seg2) {
+        let pt = l1.eval(hit.t0);
+        // the bezierTools code for line intersections has a bunch of special
+        // cases that were causing us to deviate, so we try to cover those here
+        // as they come up:
+        //
+        // special check for close x coords:
+        // https://github.com/fonttools/fonttools/blob/a6f59a4f87a011/Lib/fontTools/misc/bezierTools.py#L1193-L1212
+
+        if py_isclose(l1.end().x, l1.start().x) || py_isclose(l2.start().x, l2.end().x) {
```

Comment:
Optional, this would be easier to read done via trait (so it goes ltr), e.g. `l1.x_almost_unchanged()` or `l1.start().x.approx_eq(l1.end().x)` or some such

Reply:
I'm really just trying to clearly replicate the code that I've linked to above, which is using the `math.is_close` fn from the python std.